From 224873962ebb66d96e43228bfd042b6aa1873644 Mon Sep 17 00:00:00 2001
From: John Kim <johnykim@google.com>
Date: Thu, 31 Oct 2024 18:36:21 +0000
Subject: [PATCH] [FPF] Remove dangling ptr annotations from FingerprintingProtectionWebContentsHelper and ThrottleManager.

Remove "DanglingUntriaged" annotation from pref_service_, tracking_protection_settings_, dealer_handle_, and page_.
Reorder ThrottleManagerTest destruction order to avoid triggering phantom dangling pointer catches in tests.

Bug: 370565001
Change-Id: I7612600f927b32e787578f97d26ee25d931bf46d
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5938685
Reviewed-by: Tanushree Bisht <tanub@google.com>
Commit-Queue: John Kim <johnykim@google.com>
Reviewed-by: Martin Verde <thesalsa@google.com>
Cr-Commit-Position: refs/heads/main@{#1376488}
---

--- a/components/fingerprinting_protection_filter/browser/fingerprinting_protection_web_contents_helper.h
+++ b/components/fingerprinting_protection_filter/browser/fingerprinting_protection_web_contents_helper.h
@@ -134,14 +134,12 @@
   base::ObserverList<FingerprintingProtectionObserver>::Unchecked
       observer_list_;
 
-  // TODO(https://crbug.com/40280666): Triage dangling pointers.
-  raw_ptr<PrefService, DanglingUntriaged> pref_service_;
+  raw_ptr<PrefService> pref_service_;
 
-  raw_ptr<privacy_sandbox::TrackingProtectionSettings, DanglingUntriaged>
+  raw_ptr<privacy_sandbox::TrackingProtectionSettings>
       tracking_protection_settings_;
 
-  raw_ptr<subresource_filter::VerifiedRulesetDealer::Handle, DanglingUntriaged>
-      dealer_handle_;
+  raw_ptr<subresource_filter::VerifiedRulesetDealer::Handle> dealer_handle_;
 
   // Whether the  profile is in Incognito mode.
   bool is_incognito_;
--- a/components/fingerprinting_protection_filter/browser/throttle_manager.h
+++ b/components/fingerprinting_protection_filter/browser/throttle_manager.h
@@ -309,9 +309,7 @@
   // this class transferred onto it (in
   // FingerprintingProtectionWebContentsHelper) we'll set this member to point
   // to it.
-  //
-  // TODO(https://crbug.com/40280666): Triage dangling pointers.
-  raw_ptr<content::Page, DanglingUntriaged> page_ = nullptr;
+  raw_ptr<content::Page> page_ = nullptr;
 
   // One ThrottleManger per page means one page activation decision per throttle
   // manager.
--- a/components/fingerprinting_protection_filter/browser/throttle_manager_unittest.cc
+++ b/components/fingerprinting_protection_filter/browser/throttle_manager_unittest.cc
@@ -243,8 +243,9 @@
   }
 
   void TearDown() override {
-    test_support_.reset();
+    content::RenderViewHostTestHarness::DeleteContents();
     dealer_handle_.reset();
+    test_support_.reset();
     content::RenderViewHostTestHarness::TearDown();
   }
 
